src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -776,6 +777,12 @@ def batch_insert_into_statement(table_name: str, column_names: List[str]) -> str
         f"{COMMA.join([QUESTION_MARK] * len(column_names))}{RIGHT_PARENTHESIS}"
     )
 
+def insert_overwrite_statement(table_name: str, child: str, column_names:list) -> str:
+    column_names_str = ",".join(column_names)
+    return (
+            f"{INSERT}{OVERWRITE}{INTO}{table_name}{LEFT_PARENTHESIS}{column_names_str}{RIGHT_PARENTHESIS}"
+            f"{project_statement([], child)}"
```

Comment:
can we use similar logic of testing `is_sql_statement` as done in `insert_into_statement`. On second though, can you update `insert_into_statement` with optional argument overwrite since they have much in common?

Reply:
Great suggestion